src/JointTrajectoryControllerBase.cpp

```diff
@@ -202,10 +211,24 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time& time,
     trajectory->evaluateDerivative(timeFromStart, 2, mDesiredAcceleration);
 
     // TODO: Check path constraints.
-    // TODO: Check goal constraints.
+
+    // Check goal constraints.
+    bool goalConstraintsSatisfied = true;
+    for (const auto& dof : mControlledSkeleton->getDofs()) {
+      for (const auto& pair : mGoalConstraints) {
+        if (pair.first == dof->getName()) {
+          std::size_t index = dof->getIndexInSkeleton();
```

Comment:
This is incorrect because `getIndexInSkeleton` returns the index of the dof in the `skeleton` containing the dof, which may be different from `mControlledSkeleton` (the metaskeleton we control). 

`MetaSkeleton` support `getIndexOf(DegreeOfFreedom*)`  so you can either use that or do the for loop with `std::size_t index`

Reply:
Ohhh. So to get this straigt: A Dof belongs *to **one** Skeleton. But it multiple MetaSkeletons may reference it. The Dof has no knowledge about any MetaSkeletons, but only the one Skeleton it belongs to. Right?

Reply:
Yep, that's correct!